core/src/data/rasterSource.cpp

```diff
@@ -28,12 +31,9 @@ class RasterTileTask : public BinaryTileTask {
     }
 
     void process(TileBuilder& _tileBuilder) override {
-
-        auto source = reinterpret_cast<RasterSource*>(m_source.get());
-
         if (!m_texture) {
             // Decode texture data
-            m_texture = source->createTexture(*rawTileData);
+            m_texture = rasterSource()->createTexture(*rawTileData);
```

Comment:
This could segfault here,  `rasterSource()` can return a `nullptr` if the original `m_source` has been destroyed.

Reply:
Good point!